runtime/src/iree/hal/cts/semaphore_submission_test.h

```diff
@@ -582,8 +582,80 @@ TEST_P(semaphore_submission_test, TwoBatchesWaitingOnDifferentSemaphoreValues) {
   iree_hal_command_buffer_release(command_buffer22);
 }
 
-// TODO: test host + device -> device synchronization: submit two batches
-// with a later batch waiting on both a host and device singal to proceed.
+// Test host + device -> device synchronization: submit two batches
+// with a later batch waiting on both a host and device signal to proceed.
+TEST_P(semaphore_submission_test, BatchWaitingOnAnotherAndHostSignal) {
+  // Signal/wait relation:
+  //
+  // command_buffer1
+  //        ↓
+  //    semaphore1   semaphore2
+  //        ↓       ↙
+  // command_buffer2
+  //        ↓
+  //    semaphore3
+
+  iree_hal_command_buffer_t* command_buffer1 = CreateEmptyCommandBuffer();
+  iree_hal_command_buffer_t* command_buffer2 = CreateEmptyCommandBuffer();
+  iree_hal_semaphore_t* semaphore1 = CreateSemaphore();
+  iree_hal_semaphore_t* semaphore2 = CreateSemaphore();
+  iree_hal_semaphore_t* semaphore3 = CreateSemaphore();
+
+  uint64_t semaphore_signal_value = 1;
+
+  // Submit command_buffer2.
+  iree_hal_semaphore_t* command_buffer2_wait_semaphore_array[2] = {semaphore1,
```

Comment:
Nit: I'd just drop the explicit size (2) here.

Reply:
done